boards/BMS/Src/App/App_Eeprom.c

```diff
@@ -0,0 +1,207 @@
+#include "App_Eeprom.h"
+
+#define SOC_ADDR_PAGE (0U)
+#define SOC_ADDR_OFFSET (0U)
+#define SOC_ADDR_SIZE_BYTES (1U)
+#define NUM_PAGES (128U)
+#define SAVED_COPIES 3U
+#define BYTES_PER_FLOAT sizeof(float) / sizeof(uint8_t)
+#define BYTES_PER_SHORT sizeof(uint16_t) / sizeof(uint8_t)
+#define SHORT_SIZE ((uint8_t)sizeof(uint16_t))
+
+static void convert_float_to_bytes(uint8_t *bytes, float float_to_convert)
```

Comment:
i think that all of these conversion functions are a bit overcomplicated. also if i remember correctly, accessing unions like this isn't portable due to endianness and you're lucky that ARM and x86 are both little endian

as per https://github.com/UBCFormulaElectric/Consolidated-Firmware/pull/903#discussion_r1159114345:

you can replace convert_float_to_bytes() with:
```c
float f = ...;
const uint8_t const * bytes = (const uint8_t const *)&f; // const pointer to const int
```

and simplify convert_bytes_to_float() with:
```c
memcpy(&f, bytes, sizeof(f));
```

you can keep them in functions if you want to keep the tests

Reply:
ah good point, I overlooked this in your previous comment. I'll look into switching to this. tbh the static functions are causing just as much as a testing headache as just using these inline, might be worth abandoning.